ext/mp-rest-client/src/main/java/org/glassfish/jersey/restclient/BeanClassModel.java

```diff
@@ -0,0 +1,299 @@
+                                                   FormParam.class,
+                                                   resolveValueFromField(field, instance));
+                });
+        return form;
+    }
+
+    private Object resolveValueFromField(Field field, Object instance) {
+        try {
+            Object toReturn;
+            field.setAccessible(true);
+            toReturn = field.get(instance);
+            field.setAccessible(false);
+            return toReturn;
+        } catch (IllegalAccessException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static class Builder {
+
+        private final InterfaceModel interfaceModel;
+        private final Class<?> beanClass;
+        private ArrayList<ParamModel> parameterModels = new ArrayList<>();
+
+        private Builder(InterfaceModel interfaceModel, Class<?> beanClass) {
+            this.interfaceModel = interfaceModel;
+            this.beanClass = beanClass;
+        }
+
+        /**
+         * Parses all {@link PathParam} annotated fields from bean class.
+         *
+         * @return updated builder instance
+         */
+        Builder processPathFields() {
+            Stream.of(beanClass.getDeclaredFields())
+                    .filter(field -> field.isAnnotationPresent(PathParam.class))
+                    .forEach(field -> parameterModels.add(ParamModel.from(interfaceModel, field.getType(), field, -1)));
+            return this;
+        }
+
+        /**
+         * Parses all {@link HeaderParam} annotated fields from bean class.
+         *
+         * @return updated builder instance
+         */
+        Builder processHeaderFields() {
+            Stream.of(beanClass.getDeclaredFields())
+                    .filter(field -> field.isAnnotationPresent(HeaderParam.class))
+                    .forEach(field -> parameterModels.add(ParamModel.from(interfaceModel, field.getType(), field, -1)));
+            return this;
+        }
+
+        /**
+         * Parses all {@link CookieParam} annotated fields from bean class.
+         *
+         * @return updated builder instance
+         */
+        Builder processCookieFields() {
+            Stream.of(beanClass.getDeclaredFields())
+                    .filter(field -> field.isAnnotationPresent(CookieParam.class))
+                    .forEach(field -> parameterModels.add(ParamModel.from(interfaceModel, field.getType(), field, -1)));
+            return this;
+        }
+
+        /**
+         * Parses all {@link QueryParam} annotated fields from bean class.
+         *
+         * @return updated builder instance
+         */
+        Builder processQueryFields() {
+            Stream.of(beanClass.getDeclaredFields())
+                    .filter(field -> field.isAnnotationPresent(QueryParam.class))
+                    .forEach(field -> parameterModels.add(ParamModel.from(interfaceModel, field.getType(), field, -1)));
+            return this;
```

Comment:
make private method 
Builder processFieldsByParameterClass(Class<?> parameterClass) {
            Stream.of(beanClass.getDeclaredFields())
                    .filter(field -> field.isAnnotationPresent(parameterClass))
                    .forEach(field -> parameterModels.add(ParamModel.from(interfaceModel, field.getType(), field, -1)));
            return this;
        }

and call that private method from those 5 process*Fields() with appropriate parameter

Reply:
I have updated this builder as you suggested.